airbyte-cdk/bulk/toolkits/load-db/src/test/kotlin/io/airbyte/cdk/load/toolkits/load/db/orchestration/TableNameMapFactoryTest.kt

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.toolkits.load.db.orchestration
+
+import io.airbyte.cdk.ConfigErrorException
+import io.airbyte.cdk.load.command.Append
+import io.airbyte.cdk.load.command.DestinationCatalog
+import io.airbyte.cdk.load.command.DestinationStream
+import io.airbyte.cdk.load.data.AirbyteType
+import io.airbyte.cdk.load.data.FieldType
+import io.airbyte.cdk.load.data.ObjectType
+import io.airbyte.cdk.load.orchestration.db.ColumnNameGenerator
+import io.airbyte.cdk.load.orchestration.db.FinalTableNameGenerator
+import io.airbyte.cdk.load.orchestration.db.RawTableNameGenerator
+import io.airbyte.cdk.load.orchestration.db.TableName
+import io.airbyte.cdk.load.orchestration.db.legacy_typing_deduping.DEFAULT_AIRBYTE_INTERNAL_NAMESPACE
+import io.airbyte.cdk.load.orchestration.db.legacy_typing_deduping.TableCatalogFactory
+import io.mockk.every
+import io.mockk.mockk
+import org.junit.jupiter.api.Assertions.assertAll
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertThrows
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+
+class TableNameMapFactoryTest {
+
+    @Test
+    fun testThrowOnEmptyCatalog() {
+        val rawTableNameGenerator = mockk<RawTableNameGenerator>()
+        val finalTableNameGenerator = mockk<FinalTableNameGenerator>()
+        val columnNameGenerator = mockk<ColumnNameGenerator>()
+
+        val emptyCatalog = mockk<DestinationCatalog>()
+        every { emptyCatalog.streams } returns emptyList()
+
+        val factory =
+            TableCatalogFactory(
+                emptyCatalog,
+                rawTableNameGenerator,
+                finalTableNameGenerator,
+                columnNameGenerator
+            )
+
+        val exception = assertThrows(ConfigErrorException::class.java) { factory.get() }
+        assertTrue(exception.message!!.contains("catalog contained no streams"))
+    }
+
+    @Test
+    fun testTableNameCollision() {
+        val rawTableNameGenerator = mockk<RawTableNameGenerator>()
+        val finalTableNameGenerator = mockk<FinalTableNameGenerator>()
+        val columnNameGenerator = mockk<ColumnNameGenerator>()
+
+        // Create the same streams as in the original test - "foobarfoo" and "foofoo"
+        val stream1 = createTestStream("foobarfoo", "a")
+        val stream2 = createTestStream("foofoo", "a")
+        val catalog = DestinationCatalog(listOf(stream1, stream2))
+
+        // Mock the generators to simulate name collision by removing "bar" for both raw and final
+        // table names
+        every { rawTableNameGenerator.getTableName(stream1.descriptor) } returns
+            TableName(
+                "airbyte_internal",
+                "foofoo"
```

Comment:
nit: I think the old tests were returning `a_abab_foofoo`, not just `foofoo`. Doesn't really affect anything, but would match real behavior more closesly (i.e. the raw table name includes both the stream name + namespace, not just the name)